File: 29-06-2025/p1498.py
class Solution:
    def numSubseq(self, nums: List[int], target: int) -> int:
        mod = 10**9 + 7
        nums.sort()
        n = len(nums)
        left, right = 0, n - 1
        res = 0

        while left <= right:
            if nums[left] + nums[right] <= target:
                res += pow(2, right - left, mod)
                res %= mod
                left += 1
            else:
                right -= 1

        return res

File: 29-06-2025/p1498.php
class Solution {

    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer
     */
    function numSubseq($nums, $target) {
        sort($nums);
        $mod = 1e9 + 7;
        $n = count($nums);
        
        // Precompute powers of 2
        $pow = array_fill(0, $n, 1);
        for ($i = 1; $i < $n; $i++) {
            $pow[$i] = ($pow[$i - 1] * 2) % $mod;
        }

        $left = 0;
        $right = $n - 1;
        $res = 0;

        while ($left <= $right) {
            if ($nums[$left] + $nums[$right] <= $target) {
                $res = ($res + $pow[$right - $left]) % $mod;
                $left++;
            } else {
                $right--;
            }
        }

        return $res;
    }

}